### PR TITLE
fix(charting): remove categories added in Chart 2,  after user confirmation, when user sets addCategoryEnabled to false PD-1724

### PR DESCRIPTION
**Pull request by CarlaCostea**
https://illuminate.atlassian.net/browse/PD-1724

### CODE REVIEW

**Comment by andreeapescar** [comment]
@CarlaCostea can you please fix conflicts?

File: packages/charting/configure/src/charting-config.jsx [hunk 1]
```diff
@@ -19,6 +19,36 @@ const styles = theme => ({
...
 });
 
+const restoreCorrectAnswer = (correctAnswer, data) => {
+  if (!correctAnswer) {
+    return data;
+  }
+
+  let correctResponseDefinition = [];
+
+  data.forEach((category, currentIndex) => {
```
Comment (andreeapescar) [inline packages/charting/configure/src/charting-config.jsx:29]: Add sanity check for data.

File: packages/charting/configure/src/charting-config.jsx [hunk 2]
```diff
@@ -19,6 +19,36 @@ const styles = theme => ({
...
+    const editable = category.editable;
+    const interactive = category.interactive;
```
Comment (andreeapescar) [inline packages/charting/configure/src/charting-config.jsx:31]: You can use destructing here.

File: packages/charting/configure/src/charting-config.jsx [hunk 3]
```diff
@@ -45,7 +74,9 @@ export class ChartingConfig extends React.Component {
   changeData = data => this.props.onChange({ ...this.props.model, data });
 
   changeAddRemoveEnabled = value => {
-    const { correctAnswer } = this.state;
+    const correctAnswer  = restoreCorrectAnswer( this.props.model.correctAnswer.data,
```
Comment (andreeapescar) [inline packages/charting/configure/src/charting-config.jsx:77]: Destruct model from props

File: packages/charting/configure/src/correct-response.jsx [hunk 4]
```diff
@@ -133,9 +133,9 @@ export class CorrectResponse extends React.Component {
     const { categories } = this.state;
     let nextCategories = [];
 
-    if (nextData.length > data.length && nextCorrectAnswerData.length > data.length) {
+    if ((nextData.length > data.length ) || (nextData.length > data.length && nextCorrectAnswerData.length > data.length)) {
       nextCategories = insertCategory(nextCorrectAnswerData, nextData);
```
Comment (andreeapescar) [inline packages/charting/configure/src/correct-response.jsx:137]: Here you have A || (A && B). Why?

Comment (CarlaCostea) [reply packages/charting/configure/src/correct-response.jsx:137]: hm, I have to check why I had A&& B at the beginning

File: packages/charting/configure/src/charting-config.jsx (hunk 2 above)
Comment (CarlaCostea) [reply packages/charting/configure/src/charting-config.jsx:31]: done

File: packages/charting/configure/src/charting-config.jsx (hunk 3 above)
Comment (CarlaCostea) [reply packages/charting/configure/src/charting-config.jsx:77]: done

File: packages/charting/configure/src/charting-config.jsx (hunk 1 above)
Comment (CarlaCostea) [reply packages/charting/configure/src/charting-config.jsx:29]: done

**Pull request merged**